### ISSUE TITLE
Scrub status "unknown" - Leap 15.6 OS base

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2872]
Thanks to forum member Tex1954 for highlighting this issue. We currently report "unknown" for scrub status on a 15.6 base, but work as expected on a 15.5 base. Thanks to @FroggyFlox & @Hooverdan96 for helping to narrow down this issue.

N.B. @Hooverdan96 reports that a Stable Backport kernel does not exibity this error, this instance would also have a newer btrfs-progs userland as per: https://rockstor.com/docs/howtos/stable_kernel_backport.html

This issue is currently believed to be cosmetic in nature as the scrub itself is executed as expected. However cosmetic failure, in this context, translates to a Web-UI failure in a critical report: adding to our current Stable Milestone as a consequence.

- [x] Establish unit-test reproducer
- [x] Establish fix in the context of our existing tests and the new reproducer test.

---

Forum thread reference: https://forum.rockstor.com/t/5-0-11-scrub-not-working/9585

Seen last in 5.0.12-0

**Comment by phillxnet** [comment]
@FroggyFlox has now established the likely area/origin of this issue: hence it's creation. It looks like our scrub parser is failing over to reporting "unknown" when trying to establish a scrub status; for an as yet unknown reason. But we suspect it is down to differences in kernel versions, or our failure to apply the correct parsing accordingly. We currently have a multi-path parsing capability re scrub that adapts to the expected scrub status output: based on kernel version (from memory).

**Comment by phillxnet** [comment]
In our view we call low-level scrub_status with the Pool and the output of btrfsprogs_legacy: our designation for the appropriate parsing path for scrub_status to take.

https://github.com/rockstor/rockstor-core/blob/7639973ed10f95feddbbfff62527e13d448de503/src/rockstor/storageadmin/views/pool_scrub.py#L52-L69

And our initial value for the pool scrub status (PoolScrub.status) is established when we create the associated model used to track the subprocess we initiate the scrub via; by way of the "started" default:

https://github.com/rockstor/rockstor-core/blob/7639973ed10f95feddbbfff62527e13d448de503/src/rockstor/storageadmin/models/scrub.py#L23-L29

So the model is always initiated with scrub status assumed "started" and we then update this via a low level call to scrub_status(): but only if it is a new model (status == "started") or already established as "running".

**Comment by FroggyFlox** [comment]
> btrfsprogs_legacy

Thanks a lot for creating all of that, @phillxnet!
Your mention of btrfsprogs_legacy makes me wonder now whether we have an issue with parsing btrfs progs version... if we indeed fail to accurately detect that version, then our parsing of the scrub status command will indeed fail most likely. Your idea would fit very well in explaining what I've seen.

**Comment by phillxnet** [comment]
@FroggyFlox Thanks for kicking this one off. I'm dabbling now and should hopefully have some more idea about where we are failing in a bit.

**Comment by phillxnet** [comment]
So it seems we are getting to the nub of it here. Just got the following on a 15.6 OS base via some temporary logging:
```
[15/Jul/2024 14:25:13] INFO [fs.btrfs:1891] btrfsprogs_legacy() out=['btrfs-progs v6.5.1 ', ''], err=[''], rc=0, with btrfs_progs_version=['6', '5', '1']
[15/Jul/2024 14:25:13] INFO [fs.btrfs:1965] scrub_status_raw(mnt_pt=/mnt2/rock-pool, legacy=True) CALLED.
```
But `legacy` here is meant to be:
https://github.com/rockstor/rockstor-core/blob/7639973ed10f95feddbbfff62527e13d448de503/src/rockstor/fs/btrfs.py#L1879

With the reproducer of an "unknown" status: even after page refresh:
![unknown-scrub-status-leap-15 6](https://github.com/user-attachments/assets/d82281dd-c9f0-42dd-a758-c80bddeca7a4)

**Comment by phillxnet** [comment]
Noteworthy here is that our current 'legacy' parsing pertains to btrfs-progs "... < "v5.1.2" ...", which from the following existing unit test:
https://github.com/rockstor/rockstor-core/blob/7639973ed10f95feddbbfff62527e13d448de503/src/rockstor/fs/tests/test_btrfs.py#L1001-L1018

equates to Leap versions before 15.4 !! We are no longer publishing rpms fro anything older.

As such, it may be we can now drop this legacy test entirely.

**Comment by phillxnet** [comment]
> As such, it may be we can now drop this legacy test entirely.

I'll begin by proving the findings here and working out why we label this newer version as legacy: most likely as we inadvertently roll-over to legacy in our now dated btrfsprogs_legacy() procedure.

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2873
